### DISCUSSION TITLE
cFS's capabilities

### DISCUSSION THREAD

**Discussion by stellansarduy** [nasa/cFS#744] [Q&A]
I wanted to ask what cFS's file transfer capabilities are like and what speed does it cap out at?

**Comment by avan989** [comment]
CFS has a CF application that follows the CCSDS file delivery protocol (CFDP) version 5. As for speed, I would think it depends on your system and the configuration.

**Comment by hazytea** [comment]
The CF app chunks files up into software busmessages.  cFE has a limit on SB packet size of 32K.  Figure out the acceptable rate at which you can send packets, multiply it by 32K, and that's about the limit.  (There's a little overhead with packet headers.)

The theoritical limit is pretty high, but keep in mind system bottlenecks.  Maybe it's file write speed.  Maybe you have hardware downstream which has a smaller packet size limit.  Maybe your ground system can't handle 1000 packets/second.
